src/test/java/com/redhat/darcy/webdriver/WebDriverCookieManagerTest.java

```diff
@@ -0,0 +1,174 @@
+
+    @Before
+    public void stubDriver() {
+        when(mockDriver.manage())
+                .thenReturn(mockOptions);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameAndValue() {
+        Cookie cookie = new Cookie("chocolate", "chip");
+        org.openqa.selenium.Cookie seleniumCookie = new org.openqa.selenium.Cookie("chocolate",
+                "chip");
+
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameValueAndPath() {
+        Cookie cookie = new Cookie("chocolate", "chip", "/home/");
+        org.openqa.selenium.Cookie seleniumCookie = new org.openqa.selenium.Cookie("chocolate",
+                "chip", "/home/");
+
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameValuePathAndExpiry() {
+        Instant now = Instant.now();
+        Cookie cookie = new Cookie("chocolate", "chip", "/home/", LocalDateTime.ofInstant(now,
+                ZoneId.systemDefault()));
+        org.openqa.selenium.Cookie seleniumCookie = new org.openqa.selenium.Cookie("chocolate",
+                "chip", "/home/", Date.from(Instant.now()));
+
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameValuePathAndNullExpiry() {
+        Cookie cookie = new Cookie("chocolate", "chip", "/home/", null);
+        org.openqa.selenium.Cookie seleniumCookie = new org.openqa.selenium.Cookie("chocolate",
+                "chip", "/home/", null);
+
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameValueDomainPathAndExpiry() {
+        Instant now = Instant.now();
+        Cookie cookie = new Cookie("chocolate", "chip", "test.com", "/home/",
+                LocalDateTime.ofInstant(now, ZoneId.systemDefault()));
+        org.openqa.selenium.Cookie seleniumCookie = new org.openqa.selenium.Cookie("chocolate",
+                "chip", "test.com", "/home/", Date.from(now));
+
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldGetCookie() {
+        Instant now = Instant.now();
+        LocalDateTime localdateTime = LocalDateTime.ofInstant(now, ZoneId.systemDefault())
+                .truncatedTo(ChronoUnit.SECONDS);
+        Cookie cookie = new Cookie("chocolate", "chip", "test.com", "/home/",
+                localdateTime, true, true);
+        org.openqa.selenium.Cookie seleniumCookie = new org.openqa.selenium.Cookie("chocolate",
+                "chip", "test.com", "/home/", Date.from(now), true, true);
```

Comment:
We should have an `add` test for these boolean flags as well, I don't think that's being tested yet.